pkg/querier/distributor_queryable.go

```diff
@@ -37,13 +41,25 @@ type distributorQuerier struct {
 	distributor Distributor
 	ctx         context.Context
 	mint, maxt  int64
+	streaming   bool
+	chunkIterFn chunkIteratorFunc
 }
 
 func (q *distributorQuerier) Select(sp *storage.SelectParams, matchers ...*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
-	mint, maxt := q.mint, q.maxt
-	if sp != nil {
-		mint = sp.Start
-		maxt = sp.End
+	// Kludge: Prometheus passes nil SelectParams if it is doing a 'series' operation,
+	// which needs only metadata.
+	if sp == nil {
```

Comment:
Note to myself: this is the exact copy of `querier.metadataQuery()`, which both streaming and non-streaming `Queryable` were using before 👍